### DISCUSSION TITLE
Tunning Model Pada Decision Tree

### DISCUSSION THREAD

**Discussion by joejnt** [teamalgoritma/community#513] [Q&A] [answered]
Dear rekan2,

Saya mendapatkan permasalahan pada pembuatan model Decision Tree (DT) saat akan melakukan tuning model, dimana sebelum melakukan tuning model, hasil pemodelan pada data adalah sebagai berikut:
Data test
<img width="260" alt="image" src="https://user-images.githubusercontent.com/103427989/215009818-3fba7b2d-81b9-4dc1-8656-c7999b8829d7.png">

Data train
<img width="257" alt="image" src="https://user-images.githubusercontent.com/103427989/215009766-d956794c-6390-4ccb-80ab-88f234946ac6.png">

Dilihat dari hasilnya, model perlu dilakukan tuning dengan melakukan pruning seperti berikut:
<img width="407" alt="image" src="https://user-images.githubusercontent.com/103427989/215019256-d018a444-11ee-4b57-a916-9c223e2d92f0.png">

Namun saya mencoba untuk mencari bagaimana mendapatkan nilai dari mincriterion, minsplit dan minbucket yang optimal, dimana saya mejalankan syntax terlampir dan hasilnya error.

<img width="504" alt="image" src="https://user-images.githubusercontent.com/103427989/215019585-75a34ce3-a528-4e50-a1e5-42c73a1e3462.png">

Mohon sarannya untuk koreksi syntax diatas bagaimana, dan kalo ada syntax bagaimana cara mencari titik optimal dari mincriterion, minsplit dan minbucket agar pemodelan tidak over/underfiting.

Best regards,
JT

**Comment by dwiknrd** [comment, marked as answer]
Dear [joejnt](https://github.com/joejnt),

Pada model decision tree memang ada kecenderungan untuk terjadi overfitting. Salah satu solusinya adalah dengan melakukan regularisasi seperti pruning. Untuk mendapatkan parameter yang sesuai kita dapat mencari menggunakan beberapa metode, salah satunya adalah grid search seperti yang anda lakukan diatas. Namun error tersebut kemungkinan terjadi dikarenakan anda kurang tepat dapat memasukkan data pada parameter. Berdasarkan referensi yang didapat, anda bisa langsung mencari nilai parameter yang optimal dengan cara berikut:

```
classifier = train(form = Target ~ ., 
                   data = training_set, 
                   method = 'ctree',
                   tuneGrid = data.frame(mincriterion = seq(0.01,0.99,length.out = 100)),
                   trControl = trainControl(method = "boot",
                                            summaryFunction = defaultSummary,
                                            verboseIter = TRUE))
```

atau anda juga bisa melihat pada dokumentasi berikut:

- https://towardsdatascience.com/decision-tree-hyperparameter-tuning-in-r-using-mlr-3248bfd2d88c

Semoga dapat membantu :)

Best Regards,
Team Algoritma

**Reply by joejnt** [reply]
Dear @dwiknrd 

Terima kasih untuk insidenya, untuk syntax diatas apakah hanya untuk mencari nilai mincriterion ya ? Bagaimana dengan untuk mencari nilai dari minsplit dan minbucket.

Saya coba menjalan syntax ini, untuk yg minsplit. Namun hasilnya masih meminta untuk parameter grid harus menyebutkan mincriterion, mohon sarannya.

<img width="661" alt="image" src="https://user-images.githubusercontent.com/103427989/215472152-04a57e31-a51f-456f-90d1-9ab7936a7e53.png">


Best regards,
JT
